src/wp-includes/fonts.php

```diff
@@ -122,6 +122,17 @@ function wp_get_font_dir() {
 		'error'   => false,
 	);
 
+	wp_mkdir_p( $defaults['path'] );
+
+	if ( ! wp_is_writable( $defaults['path'] ) ) {
```

Comment:
```suggestion
	if ( ! $made_dir || ! wp_is_writable( $defaults['path'] ) ) {
```

`wp_mkdir_p()` returns true if the directory exists or could be created. If it returns false, then it's already known that the path isn't writable.